rebasehelper/build_tools/koji_tool.py

```diff
@@ -102,10 +109,27 @@ def _scratch_build(cls, source, **kwargs):
             logger.info('RPM build failed %s', weburl)
             logs.append(weburl)
             cls.logs.append(weburl)
-            raise BinaryPackageBuildError
+            cls.logs.extend(logs)
+            raise BinaryPackageBuildError(return_code=cls.get_mock_return_code(build_error))
         logs.append(weburl)
         return rpms, logs, task_id
 
+    @classmethod
+    def get_mock_return_code(cls, msg):
+        """
+        Parses koji status message for mock build return code
+
+        :param msg: mock status message
+        :return: returns mock build return code
+        """
+        found = re.search(r'mock exited with status \d+', msg)
+        if found:
+            # Example error message that can be parsed:
+            # BuildError: error building package (arch noarch),
+            # mock exited with status 1; see build.log for more information
+            return int(found.group(0).split()[-1])
```

Comment:
I have to admit this is quite original solution ;-)
But I would prefer something like this instead:
```python
match = re.search(r'mock exited with status (\d+)', msg)
if match:
    return int(match.group(1))
```

Reply:
Is it safe attempting to access a group, that is potentially non-existent?

Reply:
Yes you're right. Thanks!